src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php

```diff
@@ -180,6 +180,44 @@ public function testTransformDateTimeImmutableTimezones()
         $this->assertEquals($dateTime->format('d.m.Y, H:i'), $transformer->transform($input));
     }
 
+    public function testTransformDateTimeWithCalendar()
+    {
+        $locale = \Locale::getDefault();
+        \Locale::setDefault('en_US');
+
+        $pattern = "y-MM-dd y'w'w";
+        $dateTime = new \DateTimeImmutable('2024-03-31');
+
+        $transformer = new DateTimeToLocalizedStringTransformer(
+            null,
+            null,
+            null,
+            null,
+            \IntlDateFormatter::GREGORIAN,
+            $pattern,
+        );
+
+        $this->assertEquals('2024-03-31 2024w14', $transformer->transform($dateTime));
+
+        // ISO 8601
+        $calendar = \IntlCalendar::createInstance(null);
+        $calendar->setFirstDayOfWeek(\IntlCalendar::DOW_MONDAY);
+        $calendar->setMinimalDaysInFirstWeek(4);
+
+        $transformer = new DateTimeToLocalizedStringTransformer(
+            null,
+            null,
+            null,
+            null,
+            $calendar,
+            $pattern,
+        );
```

Comment:
we prefer long lines, maybe using named args could help? our put this on one line otherwise